src/mpi/init/init_async.c

```diff
@@ -215,27 +228,55 @@ static int get_thread_affinity(bool * apply_affinity, int **p_thread_affinity, i
 }
 #endif /* DO_ASYNC_THREAD_AFFINITY */
 
-/* called inside MPID_Init_async_thread to provide device override */
-int MPIR_Init_async_thread(void)
+static struct async_thread *find_async_thread(MPIR_Stream * stream_ptr)
 {
-    int mpi_errno = MPI_SUCCESS, thr_err;
+    struct async_thread *p = NULL;
+    while ((p = (struct async_thread *) utarray_next(async_thread_list, p))) {
+        if (p->stream_ptr == stream_ptr) {
+            break;
+        } else if (stream_ptr && stream_ptr && stream_ptr->vci == p->stream_ptr->vci) {
```

Comment:
better, but this express still starts with `stream_ptr && stream_ptr`

Reply:
Fixed - https://github.com/pmodels/mpich/blob/e5c8323d896547c8f8ebd091103120a1e7c40e68/src/mpi/init/init_async.c#L237

Reply:
👍